cg/meta/workflow/mip.py

```diff
@@ -485,3 +495,83 @@ def get_analyses_from_trailblazer(
             family=family,
             data_analysis=data_analysis,
         )
+
+    def collect_hk_data(self, bundle: str, tags: str) -> list:
+        hk_data_str = []
+        hk_data_obj = self.hk.get_files(bundle=bundle, tags=[tags])
+        for obj in hk_data_obj:
+            hk_data_str.append(obj.path)
+        return hk_data_str
+
+    def collect_files_in_folder(self, folder: str) -> list:
+        files_in_folder = os.listdir(folder)
+        return files_in_folder
+
+    def get_other_format_in_same_folder(
+        self, family_obj: models.Family, query_format: str, target_format: str
+    ) -> list:
+        redundant_target_paths = []
+        query_linked_in_hk = self.collect_hk_data(family_obj.internal_id, query_format)
+        for query in query_linked_in_hk:
+            query_folder = os.path.dirname(query)
+            query_basename = os.path.basename(query)
+            query_root = os.path.splitext(query_basename)[0]
+            files_in_query_folder = self.collect_files_in_folder(query_folder)
+            for file in files_in_query_folder:
+                if file.startswith(query_root) and file.endswith(target_format):
+                    redundant_target_paths.append(file)
+        target_paths = list(dict.fromkeys(redundant_target_paths))
+        return target_paths
+
+    def check_spring_files(self, family_obj: models.Family) -> bool:
```

Comment:
Some of these functionality might already exist in apps/crunchy-api or meta/compress-api